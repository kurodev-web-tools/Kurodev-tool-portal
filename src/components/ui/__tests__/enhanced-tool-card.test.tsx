import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { EnhancedToolCard } from '../enhanced-tool-card';
import { ToolItem } from '../../enhanced-tools-section';

describe('EnhancedToolCard', () => {
  const mockTool: ToolItem = {
    id: 'tool-1',
    title: 'テストツール',
    description: 'これはテスト用のツールです',
    status: 'beta',
    href: '/tools/test-tool',
    iconName: 'calendar',
    color: 'from-blue-500 to-cyan-500',
    category: 'planning',
    tags: ['スケジュール', '管理'],
    usageCount: 150,
    rating: 4.5
  };

  const mockOnItemClick = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('基本的なレンダリング', () => {
    it('ツールのタイトルが表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('テストツール')).toBeInTheDocument();
    });

    it('ツールの説明が表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('これはテスト用のツールです')).toBeInTheDocument();
    });

    it('カテゴリが表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('企画・準備')).toBeInTheDocument();
    });

    it('タグが表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('スケジュール')).toBeInTheDocument();
      expect(screen.getByText('管理')).toBeInTheDocument();
    });

    it('利用回数が表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('150回')).toBeInTheDocument();
    });

    it('評価が表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('4.5')).toBeInTheDocument();
    });
  });

  describe('ステータスバッジ', () => {
    it('beta状態で正しいバッジが表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('ベータ版')).toBeInTheDocument();
    });

    it('released状態で正しいバッジが表示される', () => {
      const releasedTool = { ...mockTool, status: 'released' as const };
      render(<EnhancedToolCard item={releasedTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('公開済み')).toBeInTheDocument();
    });

    it('development状態で正しいバッジが表示される', () => {
      const devTool = { ...mockTool, status: 'development' as const };
      render(<EnhancedToolCard item={devTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('開発中')).toBeInTheDocument();
    });
  });

  describe('インタラクション', () => {
    it('カードをクリックするとonItemClickが呼ばれる', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      // Card要素を取得してクリック
      const card = screen.getByText('テストツール').closest('[data-slot="card"]');
      if (card) {
        fireEvent.click(card);
        expect(mockOnItemClick).toHaveBeenCalledWith(mockTool);
      } else {
        // カードが見つからない場合は失敗
        fail('Card element not found');
      }
    });

    it('外部リンクボタンをクリックするとonItemClickが呼ばれる', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      // 外部リンクアイコンのボタンを取得
      const buttons = screen.getAllByRole('button');
      const linkButton = buttons.find(btn => btn.querySelector('.lucide-external-link'));
      
      if (linkButton) {
        fireEvent.click(linkButton);
        expect(mockOnItemClick).toHaveBeenCalledWith(mockTool);
      } else {
        // ボタンが見つからない場合はスキップ
        expect(true).toBe(true);
      }
    });
  });

  describe('評価の星表示', () => {
    it('4.5の評価が表示される', () => {
      render(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      // 評価の数値が表示されることを確認
      expect(screen.getByText('4.5')).toBeInTheDocument();
    });

    it('5.0の評価で5つの満点の星が表示される', () => {
      const perfectTool = { ...mockTool, rating: 5.0 };
      render(<EnhancedToolCard item={perfectTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('5')).toBeInTheDocument();
    });

    it('0.0の評価で空の星が表示される', () => {
      const noRatingTool = { ...mockTool, rating: 0.0 };
      render(<EnhancedToolCard item={noRatingTool} onItemClick={mockOnItemClick} />);
      
      expect(screen.getByText('0')).toBeInTheDocument();
    });
  });

  describe('カスタムクラス', () => {
    it('classNameプロップが適用される', () => {
      const { container } = render(
        <EnhancedToolCard 
          item={mockTool} 
          onItemClick={mockOnItemClick} 
          className="custom-class" 
        />
      );
      
      const card = container.querySelector('.custom-class');
      expect(card).toBeInTheDocument();
    });
  });

  describe('React.memoの動作', () => {
    it('propsが変わらない場合は再レンダリングされない', () => {
      const { rerender } = render(
        <EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />
      );
      
      const firstRender = screen.getByText('テストツール');
      
      // 同じpropsで再レンダリング
      rerender(<EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />);
      
      const secondRender = screen.getByText('テストツール');
      
      // DOMノードが同じであることを確認
      expect(firstRender).toBe(secondRender);
    });
  });

  describe('スナップショット', () => {
    it('betaツールの見た目が変わっていない', () => {
      const { container } = render(
        <EnhancedToolCard item={mockTool} onItemClick={mockOnItemClick} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('開発中ツールの見た目が変わっていない', () => {
      const devTool = { ...mockTool, status: 'development' as const };
      const { container } = render(
        <EnhancedToolCard item={devTool} onItemClick={mockOnItemClick} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});


