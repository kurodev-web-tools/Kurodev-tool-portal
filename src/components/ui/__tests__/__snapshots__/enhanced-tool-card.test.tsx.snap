// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`EnhancedToolCard スナップショット betaツールの見た目が変わっていない 1`] = `
<div
  class="text-card-foreground flex flex-col gap-6 rounded-xl border py-6 group relative bg-gray-900/30 border-gray-800 shadow-lg hover:shadow-blue-500/20 transition-all duration-300 hover:scale-[1.02] cursor-pointer"
  data-slot="card"
>
  <div
    class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 pb-3"
    data-slot="card-header"
  >
    <div
      class="flex items-start justify-between"
    >
      <div
        class="flex items-center space-x-3"
      >
        <div
          class="w-10 h-10 rounded-lg flex items-center justify-center shadow-lg group-hover:shadow-xl transition-shadow bg-gradient-to-r from-blue-500 to-cyan-500"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-calendar h-5 w-5 text-white"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2v4"
            />
            <path
              d="M16 2v4"
            />
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <path
              d="M3 10h18"
            />
          </svg>
        </div>
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="text-lg md:text-xl font-semibold text-gray-200 group-hover:text-blue-600 transition-colors leading-tight tracking-wide truncate"
            data-slot="card-title"
          >
            テストツール
          </div>
          <div
            class="flex items-center space-x-2 mt-1"
          >
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-accent [a&]:hover:text-accent-foreground text-xs text-gray-400 border-gray-600"
              data-slot="badge"
            >
              企画・準備
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-primary/90 bg-yellow-500/20 text-yellow-400 border-yellow-500/30"
              data-slot="badge"
            >
              ベータ版
            </span>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 rounded-md gap-1.5 has-[>svg]:px-2.5 opacity-0 group-hover:opacity-100 transition-opacity h-10 w-10 p-0 touch-manipulation"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-external-link h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 3h6v6"
          />
          <path
            d="M10 14 21 3"
          />
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="px-6 pt-0 space-y-3"
    data-slot="card-content"
  >
    <p
      class="text-xs md:text-sm text-gray-400 leading-relaxed line-clamp-2"
    >
      これはテスト用のツールです
    </p>
    <div
      class="flex flex-wrap gap-1"
    >
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-secondary/90 text-xs bg-gray-800/50 text-gray-300 border-gray-700"
        data-slot="badge"
      >
        スケジュール
      </span>
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-secondary/90 text-xs bg-gray-800/50 text-gray-300 border-gray-700"
        data-slot="badge"
      >
        管理
      </span>
    </div>
    <div
      class="flex items-center justify-between text-xs text-gray-500"
    >
      <div
        class="flex items-center space-x-3"
      >
        <div
          class="flex items-center space-x-1"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-users h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <path
              d="M16 3.128a4 4 0 0 1 0 7.744"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
          </svg>
          <span>
            150
            回
          </span>
        </div>
        <div
          class="flex items-center space-x-1"
        >
          <div
            class="flex items-center space-x-0.5"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400/50 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
          </div>
          <span>
            4.5
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EnhancedToolCard スナップショット 開発中ツールの見た目が変わっていない 1`] = `
<div
  class="text-card-foreground flex flex-col gap-6 rounded-xl border py-6 group relative bg-gray-900/30 border-gray-800 shadow-lg hover:shadow-blue-500/20 transition-all duration-300 hover:scale-[1.02] cursor-pointer"
  data-slot="card"
>
  <div
    class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 pb-3"
    data-slot="card-header"
  >
    <div
      class="flex items-start justify-between"
    >
      <div
        class="flex items-center space-x-3"
      >
        <div
          class="w-10 h-10 rounded-lg flex items-center justify-center shadow-lg group-hover:shadow-xl transition-shadow bg-gradient-to-r from-blue-500 to-cyan-500"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-calendar h-5 w-5 text-white"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2v4"
            />
            <path
              d="M16 2v4"
            />
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <path
              d="M3 10h18"
            />
          </svg>
        </div>
        <div
          class="flex-1 min-w-0"
        >
          <div
            class="text-lg md:text-xl font-semibold text-gray-200 group-hover:text-blue-600 transition-colors leading-tight tracking-wide truncate"
            data-slot="card-title"
          >
            テストツール
          </div>
          <div
            class="flex items-center space-x-2 mt-1"
          >
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-accent [a&]:hover:text-accent-foreground text-xs text-gray-400 border-gray-600"
              data-slot="badge"
            >
              企画・準備
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-primary/90 bg-red-500/20 text-red-400 border-red-500/30"
              data-slot="badge"
            >
              開発中
            </span>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 rounded-md gap-1.5 has-[>svg]:px-2.5 opacity-0 group-hover:opacity-100 transition-opacity h-10 w-10 p-0 touch-manipulation"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-external-link h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 3h6v6"
          />
          <path
            d="M10 14 21 3"
          />
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="px-6 pt-0 space-y-3"
    data-slot="card-content"
  >
    <p
      class="text-xs md:text-sm text-gray-400 leading-relaxed line-clamp-2"
    >
      これはテスト用のツールです
    </p>
    <div
      class="flex flex-wrap gap-1"
    >
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-secondary/90 text-xs bg-gray-800/50 text-gray-300 border-gray-700"
        data-slot="badge"
      >
        スケジュール
      </span>
      <span
        class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden [a&]:hover:bg-secondary/90 text-xs bg-gray-800/50 text-gray-300 border-gray-700"
        data-slot="badge"
      >
        管理
      </span>
    </div>
    <div
      class="flex items-center justify-between text-xs text-gray-500"
    >
      <div
        class="flex items-center space-x-3"
      >
        <div
          class="flex items-center space-x-1"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-users h-3 w-3"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <path
              d="M16 3.128a4 4 0 0 1 0 7.744"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
          </svg>
          <span>
            150
            回
          </span>
        </div>
        <div
          class="flex items-center space-x-1"
        >
          <div
            class="flex items-center space-x-0.5"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="lucide lucide-star h-3 w-3 fill-yellow-400/50 text-yellow-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
          </div>
          <span>
            4.5
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
